lib/collection.dart

```diff
@@ -77,3 +77,23 @@ bool setsEqual(Set a, Set b) {
 
   return a.containsAll(b);
 }
+
+/// Returns the index of the first item in [elements] where [predicate]
+/// evaluates to true.
+///
+/// Returns -1 if there are no items where [predicate] evaluates to true.
+int indexOf<T>(Iterable<T> elements, bool predicate(T element)) {
+  if (elements is List) {
+    for (int i = 0; i < elements.length; i++) {
+      if (predicate(elements[i])) return i;
```

Comment:
The analyzer has had different behaviour over time with that. A manual run of the analyzer results in:
```
[error] The operator '[]' isn't defined for the class 'Iterable<T>'. (/home/travis/build/google/quiver-dart/lib/collection.dart, line 88, col 29)
```